### PR TITLE
Ignore commented lines that begin with a hash

### PR DESCRIPTION
**Pull request by jreedcode**
Hi Jeff,

Please review my change to allow commented lines to be ignored during icdiff.

Thanks!

### CODE REVIEW

**Comment by Jwink3101** [comment]
Two things. First, conceptually you would want to allow for comment lines to be specified with a character. The tool is not python specific. 

Second, implementation wise, this will mess up line numbering.

File: icdiff [hunk 1]
```diff
@@ -655,6 +660,12 @@ def diff_files(options, a, b):
         lines_a = lines_a[:head]
         lines_b = lines_b[:head]
 
+    if options.matcher:
+        lines_a = [line_a for line_a in lines_a if not
+                   re.search(options.matcher, line_a)]
```
Comment (jeffkaufman) [inline icdiff]: nit: I'd find this clearer with the `not` on the next line

**Comment by jeffkaufman** [comment]
Looks good!  Could you add some tests, and document that this option isn't compatible with line numbers?

**Comment by jreedcode** [comment]
OK, changes amended.

**Comment by jreedcode** [comment]
@jeffkaufman, Just a friendly nudge :) no rush.
Just since I think this is complete, and excited to see it go live.

**Comment by jeffkaufman** [comment]
Looks like the test failure is unrelated to your change (python nightly added a deprecation warning I need to look into)

**Pull request merged**